client/app/services/query.js

```diff
@@ -429,7 +429,8 @@ class Parameters {
 
     const parameterExists = p => includes(parameterNames, p.name);
     const parameters = this.query.options.parameters;
-    this.query.options.parameters = parameters.filter(parameterExists).map(p => new Parameter(p, this.query.id));
+    this.query.options.parameters = parameters.filter(parameterExists)
+      .map(p => (p instanceof Parameter ? p : new Parameter(p, this.query.id)));
```

Comment:
This was introduced a long ago in #1649, the change seems not be affecting the fix :stuck_out_tongue_closed_eyes: